--- pmdk-1.6.orig/src/test/obj_action/memcheck1.log.match
+++ pmdk-1.6/src/test/obj_action/memcheck1.log.match
@@ -49,5 +49,4 @@ $(OPT)==$(N)==    still reachable: 0 byt
 $(OPX)==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 $(OPT)==$(N)==
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 4 errors from 4 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/obj_constructor/memcheck1.log.match
+++ pmdk-1.6/src/test/obj_constructor/memcheck1.log.match
@@ -61,5 +61,4 @@ $(OPT)==$(N)==         suppressed: $(NC)
 $(OPT)==$(N)== Reachable blocks (those to which a pointer was found) are not shown.
 $(OPT)==$(N)== To see them, rerun with: --leak-check=full --show-leak-kinds=all
 $(OPT)==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/obj_ctl_debug/memcheck1.log.match
+++ pmdk-1.6/src/test/obj_ctl_debug/memcheck1.log.match
@@ -15,6 +15,5 @@
 ==$(N)== 
 ==$(N)== All heap blocks were freed -- no leaks are possible
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== Use --track-origins=yes to see where uninitialised values come from
 ==$(N)== ERROR SUMMARY: $(N) errors from $(N) contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/obj_locks/drd1.log.match
+++ pmdk-1.6/src/test/obj_locks/drd1.log.match
@@ -5,5 +5,4 @@
 ==$(N)== Parent PID: $(N)
 ==$(N)==$(W)
 ==$(N)==$(W)
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/obj_locks/helgrind2.log.match
+++ pmdk-1.6/src/test/obj_locks/helgrind2.log.match
@@ -5,7 +5,6 @@
 ==$(N)== Parent PID: $(N)
 ==$(N)==$(W)
 ==$(N)==$(W)
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== Use --history-level=approx or =none to gain increased speed, at
 ==$(N)== the cost of reduced accuracy of conflicting-access information
 ==$(N)== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/obj_memcheck/memcheck0.log.match
+++ pmdk-1.6/src/test/obj_memcheck/memcheck0.log.match
@@ -131,6 +131,5 @@ $(OPT)==$(N)==      possibly lost: 0 byt
 $(OPT)==$(N)==    still reachable: 0 bytes in 0 blocks
 $(OPT)==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== Use --track-origins=yes to see where uninitialised values come from
 ==$(N)== ERROR SUMMARY: 10 errors from 8 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/unittest/unittest.sh
+++ pmdk-1.6/src/test/unittest/unittest.sh
@@ -663,6 +663,21 @@ function valgrind_ignore_warnings() {
 }
 
 #
+# valgrind_ignore_messages -- cuts off Valgrind messages that are irrelevant
+#	to the correctness of the test, but changes during Valgrind rebase
+#	usage: valgrind_ignore_messages <log-file>
+#
+function valgrind_ignore_messages() {
+	if [ -e "$1.match" ]; then
+		cat $1 | grep -v \
+			-e "For lists of detected and suppressed errors, rerun with: -s" \
+			-e "For counts of detected and suppressed errors, rerun with: -v" \
+			>  $1.tmp
+		mv $1.tmp $1
+	fi
+}
+
+#
 # get_trace -- return tracing tool command line if applicable
 #	usage: get_trace <check type> <log file> [<node>]
 #
@@ -870,11 +885,13 @@ function expect_normal_exit() {
 			do
 				local log_file=node\_$node\_$VALGRIND_LOG_FILE
 				valgrind_ignore_warnings $new_log_file
+				valgrind_ignore_messages $new_log_file
 				validate_valgrind_log $new_log_file
 			done
 		else
 			if [ -f $VALGRIND_LOG_FILE ]; then
 				valgrind_ignore_warnings $VALGRIND_LOG_FILE
+				valgrind_ignore_messages $VALGRIND_LOG_FILE
 				validate_valgrind_log $VALGRIND_LOG_FILE
 			fi
 		fi
--- pmdk-1.6.orig/src/test/vmem_valgrind/memcheck2.log.match
+++ pmdk-1.6/src/test/vmem_valgrind/memcheck2.log.match
@@ -21,5 +21,4 @@ $(OPT)==$(N)==    by 0x$(X): vmem_malloc
 ==$(N)==    still reachable: 0 bytes in 0 blocks
 ==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/vmem_valgrind/memcheck3.log.match
+++ pmdk-1.6/src/test/vmem_valgrind/memcheck3.log.match
@@ -21,5 +21,4 @@ $(OPT)==$(N)==    by 0x$(X): vmem_malloc
 ==$(N)==    still reachable: 0 bytes in 0 blocks
 ==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/vmem_valgrind/memcheck4.log.match
+++ pmdk-1.6/src/test/vmem_valgrind/memcheck4.log.match
@@ -24,5 +24,4 @@ $(OPT)==$(N)==      possibly lost: 0 byt
 $(OPT)==$(N)==    still reachable: 0 bytes in 0 blocks
 $(OPT)==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/vmem_valgrind_region/memcheck0.log.match
+++ pmdk-1.6/src/test/vmem_valgrind_region/memcheck0.log.match
@@ -19,5 +19,4 @@ $(OPT)==$(N)==      possibly lost: 0 byt
 $(OPT)==$(N)==    still reachable: 0 bytes in 0 blocks
 $(OPT)==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 $(OPT)==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 0 errors from 0 contexts $(*)
--- pmdk-1.6.orig/src/test/vmem_valgrind_region/memcheck1.log.match
+++ pmdk-1.6/src/test/vmem_valgrind_region/memcheck1.log.match
@@ -22,5 +22,4 @@ $(OPT)==$(N)==    by 0x$(X): vmem_malloc
 ==$(N)==    still reachable: 0 bytes in 0 blocks
 ==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/vmem_valgrind_region/memcheck2.log.match
+++ pmdk-1.6/src/test/vmem_valgrind_region/memcheck2.log.match
@@ -38,6 +38,5 @@ $(OPT)==$(N)==    by 0x$(X): vmem_malloc
 ==$(N)==    still reachable: 0 bytes in 0 blocks
 ==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== Use --track-origins=yes to see where uninitialised values come from
 ==$(N)== ERROR SUMMARY: $(N) errors from 2 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/vmem_valgrind_region/memcheck3.log.match
+++ pmdk-1.6/src/test/vmem_valgrind_region/memcheck3.log.match
@@ -31,6 +31,5 @@ $(OPT)==$(N)==    by 0x$(X): vmem_malloc
 ==$(N)==    still reachable: 0 bytes in 0 blocks
 ==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 $(OPT)==$(N)== Use --track-origins=yes to see where uninitialised values come from
 ==$(N)== ERROR SUMMARY: 9 errors from 2 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/vmem_valgrind_region/memcheck4.log.match
+++ pmdk-1.6/src/test/vmem_valgrind_region/memcheck4.log.match
@@ -47,6 +47,5 @@ $(OPT)==$(N)==    by 0x$(X): vmem_malloc
 ==$(N)==    still reachable: 0 bytes in 0 blocks
 ==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 $(OPT)==$(N)== Use --track-origins=yes to see where uninitialised values come from
 ==$(N)== ERROR SUMMARY: $(N) errors from 3 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/vmmalloc_valgrind/memcheck1.log.match
+++ pmdk-1.6/src/test/vmmalloc_valgrind/memcheck1.log.match
@@ -21,5 +21,4 @@ $(OPT)==$(N)==    by 0x$(X): malloc $(*)
 ==$(N)==    still reachable: 0 bytes in 0 blocks
 ==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 ==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: $(N) from $(N))
--- pmdk-1.6.orig/src/test/vmmalloc_valgrind/memcheck2.log.match
+++ pmdk-1.6/src/test/vmmalloc_valgrind/memcheck2.log.match
@@ -24,5 +24,4 @@ $(OPT)==$(N)==      possibly lost: 0 byt
 $(OPT)==$(N)==    still reachable: 0 bytes in 0 blocks
 $(OPT)==$(N)==         suppressed: $(NC) bytes in $(N) blocks
 $(OPT)==$(N)== 
-==$(N)== For counts of detected and suppressed errors, rerun with: -v
 ==$(N)== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: $(N) from $(N))
